GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+                return;
+            }
+
+            if (Module.InTheMiddleOfBisect())
+            {
+                await SetStateAsync(State.Bisect, false);
+                return;
+            }
+
+            bool conflicted = Module.InTheMiddleOfConflictedMerge();
+
+            if (Module.InTheMiddleOfRebase())
+            {
+                await SetStateAsync(State.Rebase, conflicted);
+                return;
+            }
+
+            if (Module.InTheMiddleOfMerge())
+            {
+                await SetStateAsync(State.Merge, conflicted);
+                return;
+            }
+
+            if (Module.InTheMiddleOfPatch())
+            {
+                await SetStateAsync(State.Patch, conflicted);
+                return;
+            }
+
+            await SetStateAsync(State.None, false);
+        }
+
+        private async Task SetStateAsync(State state, bool conflicted)
+        {
+            if ((state == _state) && (conflicted == _hasConflicts))
+            {
+                // nothing to do
+                return;
+            }
+
+            _state = state;
+            _hasConflicts = conflicted;
+
+            // now update in main UI thread
+            await this.SwitchToMainThreadAsync();
+
+            // remove old controls
+            ButtonContainer.Controls.Clear();
+
+            if (_state == State.None)
+            {
+                Hide();
+                return;
+            }
+
+            BackColor = _hasConflicts ? System.Drawing.Color.DarkOrange : System.Drawing.Color.LightSkyBlue;
+
+            string action = "";
+
+            switch (_state)
+            {
+                case State.Bisect:
+                    action = _bisect.Text;
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case State.Rebase:
+                    action = _rebase.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case State.Merge:
+                    action = _merge.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    break;
+                case State.Patch:
+                    action = _patch.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+            }
+
+            Label.Text = string.Format(
+                _hasConflicts ?
+                    _progressWithConflictsMessage.Text :
+                    _progressMessage.Text,
+                action);
+
+            Show();
```

Comment:
Instead of `Show/Hide` you can use `Visible = true/false`

Reply:
I tend to prefer functional forms rather than accessing variables directly, but happy to change it if that's the standard here.

Reply:
`Visible` is a property, so it incapsulates variables.

If you look at the code, [`Show()` calls `Visilbe = true`](https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs#L11378-L11381), and [`Hide()` calls `Visible = false`](https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs#L6214-L6217).